test/cpp/api/dataloader.cpp

```diff
@@ -59,6 +59,87 @@ TEST(DataTest, TransformCallsGetApplyCorrectly) {
   ASSERT_EQ(batch, expected);
 }
 
+struct DummyChunkDataset : public datasets::ChunkDataSet<
+                               DummyChunkDataset,
+                               std::vector<int>,
+                               samplers::SequentialSampler,
+                               samplers::SequentialSampler> {
+public:
+  using BatchType = std::vector<int>;
+  using ChunkSamplerType = samplers::SequentialSampler;
+  using ExampleSamplerType = samplers::SequentialSampler;
+
+  DummyChunkDataset(size_t prefetch_count)
+      : datasets::ChunkDataSet<
+            DummyChunkDataset,
+            BatchType,
+            ChunkSamplerType,
+            ExampleSamplerType>(prefetch_count){};
+
+  /// dummy getter
+  ChunkSamplerType get_chunk_sampler() override {
+    return ChunkSamplerType(0);
+  };
+
+  /// dummy getter
+  ExampleSamplerType get_example_sampler() override {
+    return ExampleSamplerType(0);
+  };
+
+  size_t get_chunk_count() override {
+    return chunk_size_;
+  }
+
+  BatchType read_chunk(size_t chunk_index) override {
+    BatchType batchData;
+    int start_index = chunk_index == 0
+        ? 0
+        : std::accumulate(chunkSize, chunkSize + chunk_index, 0);
+
+        batchData.resize(chunkSize[chunk_index]);
+
+    std::iota(batchData.begin(), batchData.end(), start_index);
+
+    return batchData;
+  }
+
+  const static size_t chunk_size_ = 3;
+  size_t chunkSize[chunk_size_] = {10, 5, 20};
+};
+
+TEST(DataTest, ChunkDataSetGetBatch) {
+  const size_t prefetch_counts[] = {1, 2, 3, 4};
+  const size_t batch_sizes[] = {5, 7};
+
+  for (auto prefetch_count : prefetch_counts) {
+    for (auto batch_size : batch_sizes) {
+      datasets::SharedBatchDataset<DummyChunkDataset> dataset =
+          datasets::make_shared_dataset<DummyChunkDataset>(prefetch_count);
+
+      auto data_loader = torch::data::make_chunk_data_loader(
+          dataset,
+          DataLoaderOptions(batch_size).workers(0).chunk_loading(true));
+
+      dataset->reset();
+
+      std::vector<bool> result(35, false);
+
+      auto iterator = data_loader->begin();
+      for (size_t i = 0; i < (35 / batch_size); ++i, ++iterator) {
```

Comment:
(35 / batch_size) is a temporarily hack due to a bug in dataloader. For chunk data set, the data loader doesn't know when the iterator reaches to the end (i.e. equal to data_loader->end()). 

The fix will be checked in as a different PR. For now, to unblock the test, I am just hard-coding the iteration count. Once the fix is in, I'll integrate that part and remove this hack.